api/v1alpha5/virtualmachine_types.go

```diff
@@ -58,6 +58,32 @@ const (
 	VirtualMachineClassConfigurationSynced = "VirtualMachineClassConfigurationSynced"
 )
 
+const (
+	// VirtualMachineSnapshotRevertSucceeded indicates that the VM
+	// has been reverted to a snapshot.
+	VirtualMachineSnapshotRevertSucceeded = "VirtualMachineSnapshotRevertSucceeded"
+
+	// VirtualMachineSnapshotRevertInProgressReason indicates that the
+	// revert operation is in progress.
+	VirtualMachineSnapshotRevertInProgressReason = "VirtualMachineSnapshotRevertInProgress"
+
+	// VirtualMachineSnapshotRevertTaskFailedReason indicates that the
+	// revert operation is invalid.
+	VirtualMachineSnapshotRevertTaskFailedReason = "VirtualMachineSnapshotRevertTaskFailed"
+
+	// VirtualMachineSnapshotRevertFailedInvalidYAMLReason indicates
+	// that the revert operation has failed due to invalid VM spec to revert to.
+	VirtualMachineSnapshotRevertFailedInvalidYAMLReason = "VirtualMachineSnapshotRevertFailedInvalidYAML"
```

Comment:
Can you change this to somthing generic?  Like "InvalidVMManifest"?  We are talking about changing the stored YAML to JSON and I don't want the Condition to be affected by that change.